### ISSUE TITLE
Out of memory if logging gradients with HalfTensor

### ISSUE THREAD

**Issue by reymondzzzz** [wandb/client#1651]
I train big model and want logging gradients. I use FP16 in torch. One layer has 180kk params. I think that reason is https://github.com/wandb/client/blob/58405877a36beaba556cce31de304527b5245aa6/wandb/wandb_torch.py#L168-L170
Tensor was coped and upconvert in Float on gpu memory. Comment is not actual. Now HalfTensor can be used on cpu.
https://pytorch.org/docs/stable/tensors.html

`
tensor = tensor.cpu().clone().type(torch.FloatTensor).detach()
`

**Comment by cvphelps** [comment]
Thanks for this suggestion. Could you share a quick script we could use to repro?

**Comment by sydholl** [comment]
In the past year we've majorly reworked the CLI and UI for Weights & Biases. We're closing issues older than 6 months. Please comment to reopen.
